src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,514 @@
+      );
+    }
+  }
+
+  private validateDiskIops(storageCapacityGiB: number, diskIops?: number, haPairs: number = 1): void {
+    if (
+      diskIops == null ||
+      Token.isUnresolved(diskIops) ||
+      Token.isUnresolved(storageCapacityGiB) ||
+      Token.isUnresolved(haPairs)
+    ) {
+      return;
+    }
+
+    const minDiskIops = storageCapacityGiB * 3 * haPairs;
+    const maxDiskIops = 200_000 * haPairs;
+
+    if (!Number.isInteger(diskIops) || diskIops < minDiskIops || diskIops > maxDiskIops) {
+      throw new Error(`\'diskIops\' must be an integer between ${minDiskIops} and ${maxDiskIops}, got ${diskIops}`);
+    }
+  }
+
+  private validateEndpointIpAddressRange(deploymentType: OntapDeploymentType, endpointIpAddressRange?: string): void {
+    if (endpointIpAddressRange == null || Token.isUnresolved(endpointIpAddressRange)) {
+      return;
+    }
+    if (deploymentType !== OntapDeploymentType.MULTI_AZ_1 && deploymentType !== OntapDeploymentType.MULTI_AZ_2) {
+      throw new Error("'endpointIpAddressRange' can only be specified for multi-AZ file systems");
+    }
+    if (!/^[^\u0000\u0085\u2028\u2029\r\n]{9,17}$/.test(endpointIpAddressRange)) {
+      throw new Error(
+        "'endpointIpAddressRange' must be between 9 and 17 characters long and not contain any of the following characters: \\u0000, \\u0085, \\u2028, \\u2029, \\r, or \\n",
+      );
+    }
+  }
+
+  private validateFsxAdminPassword(fsxAdminPassword?: string): void {
+    if (fsxAdminPassword == null || Token.isUnresolved(fsxAdminPassword)) {
+      return;
+    }
+    if (!/^[^\u0000\u0085\u2028\u2029\r\n]{8,50}$/.test(fsxAdminPassword)) {
+      throw new Error(
+        "'fsxAdminPassword' must be between 8 and 50 characters long and not contain any of the following characters: \\u0000, \\u0085, \\u2028, \\u2029, \\r, or \\n",
+      );
+    }
+    // must contain at least one English letter and one number, and must not contain the word 'admin'.
+    if (!/[a-zA-Z]/.test(fsxAdminPassword) || !/\d/.test(fsxAdminPassword) || /admin/i.test(fsxAdminPassword)) {
+      throw new Error(
+        "'fsxAdminPassword' must contain at least one English letter and one number, and must not contain the word 'admin'",
+      );
+    }
+  }
+
+  private validateHaPairs(deploymentType: OntapDeploymentType, haPairs?: number): void {
+    if (haPairs == null || Token.isUnresolved(haPairs)) {
+      return;
+    }
+    if (!Number.isInteger(haPairs) || haPairs < 1 || haPairs > 12) {
+      throw new Error(`\'haPairs\' must be an integer between 1 and 12, got ${haPairs}`);
+    }
+    if (
+      haPairs > 1 &&
+      [OntapDeploymentType.SINGLE_AZ_1, OntapDeploymentType.MULTI_AZ_1, OntapDeploymentType.MULTI_AZ_2].includes(
+        deploymentType,
+      )
+    ) {
+      throw new Error(`\'haPairs\' must be 1 for deployment type ${deploymentType}`);
+    }
+  }
```

Comment:
Multiple HA pairs are only supported for `SINGLE_AZ_2` deployment type.

https://docs.aws.amazon.com/fsx/latest/ONTAPGuide/adding-HA-pairs.html